src/Libraries/Microsoft.Extensions.Diagnostics.Testing/Logging/FakeLogCollector.cs

```diff
@@ -129,13 +198,153 @@ internal void AddRecord(FakeLogRecord record)
             return;
         }
 
+        List<Waiter>? waitersToWakeUp;
+
         lock (_records)
         {
             _records.Add(record);
+
+            Span<bool> waitersToWakeUpOrderedByIndices = stackalloc bool[_waiters.Count];
```

Comment:
This can cause a `StackOverflowException` at runtime if the number of waiters is too high